azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/util/DeploymentTemplateBuilder.java

```diff
@@ -94,8 +91,8 @@
    private static final String DEFAULT_SUBNET_ADDRESS_PREFIX = "10.0.0.0/24";
 
    @Inject
-   DeploymentTemplateBuilder(Json json, @Assisted("group") String group, @Assisted("name") String name, @Assisted Template template,
-                             final AzureComputeServiceContextModule.AzureComputeConstants azureComputeConstants) {
+   DeploymentTemplateBuilder(Json json, @Assisted("group") String group, @Assisted("name") String name,
+         @Assisted Template template, Function<Template, LoginCredentials> templateToLoginCredentials) {
```

Comment:
jclouds already has [this function](https://github.com/jclouds/jclouds/blob/master/compute/src/main/java/org/jclouds/compute/functions/DefaultCredentialsFromImageOrOverridingCredentials.java). It takes the image default credentials (the ones from the properties) and overrides them with the values provided by users in the options. The resulting object is the right LoginCredentials, so it is worth using it here instead of exposing a shared credentials object.